### PR TITLE
Farewell dcm2nii

### LINKED ISSUE #235: Migrate dcm2nii calls to dcm2niix
`dcm2nii` is considered deprecated upstream and superseded by `dcm2niix` which is actively maintained. Besides, it's becoming very difficult to find a working binary release for OSX and it will likely never be ported to the new M1.  

See #234 for further context.

### PR DESCRIPTION
**Pull request by ghisvail**
Closes #235 

- [x] Local test on ADNI
- [x] Local test on AIBL

### CODE REVIEW

**Comment by ghisvail** [comment]
No issues with neither AIBL nor ADNI locally on a sample of a dozen of subjects with mixed sessions and cohorts.

**Comment by MatthieuJoulot** [comment]
Exemple of a print of stdout I've had: 
2021-07-29 13:17:21,793:WARNING:WARNING: errors may have occured.
Output:Compression will be faster with 'pigz' installed http://macappstore.org/pigz/
Chris Rorden's dcm2niiX version v1.0.20210317  Clang12.0.0 x86-64 (64-bit MacOS)
Found 8549 DICOM file(s)
Slice positions repeated, but number of slices (8549) not divisible by number of repeats (179): missing images?
Warning: Seconds between volumes varies
Warning: Interslice distance varies in this volume (incompatible with NIfTI format).
Unable to equalize slice distances: slice order not consistently ascending.
First spatial position repeated 179 times
Error:Error:  Recompiling with '-DmyInstanceNumberOrderIsNotSpatial' might help.

File: clinica/iotools/converters/adni_to_bids/adni_utils.py [hunk 1]
```diff
@@ -1166,7 +1166,12 @@ def create_file(image, modality, total, bids_dir, mod_to_update):
 
         if image.Is_Dicom:
             command = f"dcm2niix -b {generate_json} -z {zip_image} -o {output_path} -f {output_filename} {image_path}"
-            subprocess.run(command, shell=True, stdout=subprocess.DEVNULL)
+            output_dcm2niix = subprocess.run(command, shell=True, capture_output=True)
+            if output_dcm2niix.returncode != 0:
+                cprint(
+                    msg=f'WARNING: errors may have occured.\nOutput:{output_dcm2niix.stdout.decode("utf-8")}\nError:{output_dcm2niix.stderr.decode("utf-8")}',
+                    lvl="warning",
+                )
```
Comment (ghisvail) [inline clinica/iotools/converters/adni_to_bids/adni_utils.py:1174]: We should probably factor this part out since it's used verbatim in 3 different places

Comment (ghisvail) [reply clinica/iotools/converters/adni_to_bids/adni_utils.py:1174]: Also you can drop the `WARNING` prefix in the message. The logger will take care of it.

Comment (MatthieuJoulot) [reply clinica/iotools/converters/adni_to_bids/adni_utils.py]: Done !

**Comment by ghisvail** [comment]
Looks good to me. CI seems unable to process this PR on Linux for reasons beyond the scope of the code.

**Comment by ghisvail** [comment]
Let's go

**Pull request merged**